libkpod/ps.go

```diff
@@ -0,0 +1,189 @@
+package libkpod
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/kubernetes-incubator/cri-o/oci"
+	"github.com/pkg/errors"
+)
+
+// FilterParamsPS contains the filter options for ps
+type FilterParamsPS struct {
+	id       string
+	label    string
+	name     string
+	exited   int32
+	status   string
+	ancestor string
+	before   time.Time
+	since    time.Time
+	volume   string
+}
+
+// ParseFilter takes a filter string and a list of containers and filters it
+func ParseFilter(filter string, server *ContainerServer) (*FilterParamsPS, error) {
+	containers := server.ListContainers()
+	params := new(FilterParamsPS)
+	allFilters := strings.Split(filter, ",")
+
+	for _, param := range allFilters {
+		pair := strings.SplitN(param, "=", 2)
+		switch strings.TrimSpace(pair[0]) {
+		case "id":
+			params.id = pair[1]
+		case "label":
+			params.label = pair[1]
+		case "name":
+			params.name = pair[1]
+		case "exited":
+			exitedCode, err := strconv.ParseInt(pair[1], 10, 32)
+			if err != nil {
+				return nil, errors.Errorf("exited code out of range %q", pair[1])
```

Comment:
`exited` is a parameter for the filter, and this is where the filter params and their values are being add to the `FilterParamsPS` struct, so isn't this where the error check should take place?